### PR TITLE
[Sparse] APPNP example

### PR DESCRIPTION
**Pull request by mufeili**
## Description
APPNP example, the accuracy is > 0.82

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/sparse/appnp.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            nn.Linear(in_size, hidden_size),
+            nn.ReLU(),
+            nn.Dropout(dropout),
+            nn.Linear(hidden_size, out_size),
+        )
+        self.num_hops = num_hops
+        self.A_dropout = nn.Dropout(dropout)
+        self.alpha = alpha
+
+    def forward(self, A, X):
```
Comment (mufeili) [inline examples/sparse/appnp.py:37]: We might want to extend `DropEdge` for sparse matrix. @jermainewang

File: examples/sparse/appnp.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    D_hat = diag(A_hat.sum(dim=1)) ** -0.5
+    A_hat = D_hat @ A_hat @ D_hat
+
+    # Create APPNP model.
+    X = g.ndata["feat"]
+    in_size = X.shape[1]
+    out_size = dataset.num_classes
+    model = APPNP(in_size, out_size).to(dev)
+
+    # Kick off training.
```
Comment (frozenbugs) [inline examples/sparse/appnp.py:115]: Let's mention that the A_hat will be updated in place in the comment.

Comment (mufeili) [reply examples/sparse/appnp.py:116]: Instead of leaving a comment here, I left a comment in `APPNP.forward`.

Comment (frozenbugs) [reply examples/sparse/appnp.py:116]: Let's add the following comment at *every* place that we pass in A_hat.

# Note that A_hat will be update inside the method.

**Review by frozenbugs — APPROVED** [review]

File: examples/sparse/appnp.py (hunk 2 above)
Comment (mufeili) [reply examples/sparse/appnp.py:116]: `A_hat` does not get updated right? Because we reset `A_hat.val` to its initial value in the end.

Comment (frozenbugs) [reply examples/sparse/appnp.py:116]: Ah, I see.

File: examples/sparse/appnp.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            nn.Linear(in_size, hidden_size),
+            nn.ReLU(),
+            nn.Dropout(dropout),
+            nn.Linear(hidden_size, out_size),
+        )
+        self.num_hops = num_hops
+        self.A_dropout = nn.Dropout(dropout)
+        self.alpha = alpha
+
+    def forward(self, A, X):
```
Comment (frozenbugs) [inline examples/sparse/appnp.py:37]: A -> A_hat
